File: examples/async/kraken_ws_public.py
"""Read Kraken public WebSocket trade events."""

import asyncio

from dcex.ws import kraken


async def main() -> None:
    """Subscribe to Kraken spot trades and print two messages."""
    async with kraken.public() as ws:
        await ws.subscribe_trades("BTC-USD-SPOT")
        print(await ws.recv())
        print(await ws.recv())


if __name__ == "__main__":
    asyncio.run(main())

File: crates/dcex/examples/kraken_ws_public.rs
use std::time::Duration;

use dcex::ws::kraken::KrakenPublicWebSocket;

#[tokio::main]
async fn main() -> dcex::Result<()> {
    let mut ws = KrakenPublicWebSocket::new(Duration::from_secs(10))?;
    ws.connect().await?;
    ws.subscribe_trades("BTC-USD-SPOT").await?;
    println!("{}", ws.recv().await?);
    println!("{}", ws.recv().await?);
    ws.close().await?;
    Ok(())
}
